storage/blob.go

```diff
@@ -52,8 +87,40 @@ type Blobs interface {
 	Create(ctx context.Context, ref BlobRef, size int64) (BlobWriter, error)
 	// Open opens a reader with the specified namespace and key
 	Open(ctx context.Context, ref BlobRef) (BlobReader, error)
+	// OpenSpecific opens a reader for the already-located blob, avoiding the potential need
+	// to check multiple storage formats to find the blob.
+	OpenSpecific(ctx context.Context, ref BlobRef, formatVer FormatVersion) (BlobReader, error)
```

Comment:
This is not a blocking issue, but I would prefer a slightly more "specific" (ha...) name for these functions. Something like `OpenSpecificVer` would would imo. That way it is clear what we're being "specific" about, and how it differs from the regular Open call.

Reply:
I would also like a better name, but `OpenSpecificVer` or `OpenSpecificVersion` both suggest to me that they're talking about a blob version, rather than a storage format version. We don't have blob versions now, but they might not be an unreasonable thing to have at some point. So it seems potentially more confusing than `OpenSpecific`. At least in the `OpenSpecific` case, there isn't an obvious assumption to make about what the specificity is about, and one would be led to compare with the vanilla `Open` call to see what's different. As an alternative, I like `OpenAt` best, but it might be confusing in relation to the `ReadAt` and `WriteAt` methods on file-like objects that implement `io.ReaderAt` and `io.WriterAt`. Any other suggestions? `OpenSpecificFormat` is my next top choice, I suppose.

Reply:
A few ideas:

`OpenFormat`
`OpenSpecFormat`
`OpenSpecFmt`

:man_shrugging:

Reply:
Egon is sad when we use abbreviations in variable names.

Since this isn't meant to be used terribly often, a longer name should be fine. I'd go all the way to `OpenSpecificStorageFormatVersion` if it makes everybody happy.

Reply:
`OpenWith` or `OpenWithVersion`?

Reply:
Yes. I mentioned it elsewhere, but `Format` isn't quite the right term; it implies that it only affects the layout of bytes in a file, which is not the case. `Version` without qualification seems _very_ wrong, since we use that term for a few dozen other things already, and even with the package name, `storage.Version` could easily be mistaken as suggesting support for multiple versions of blobs. I think "storage format" is a necessary term to describe the multiple possible combinations of storage+location+fileformat, and "storage format version" is necessary to discuss standardized supported points in that space.

I could be convinced otherwise, but the replacement would need to be really good.

Reply:
I think most names are never specific enough, the question would it confuse someone when reading the code. Yes, `Version` at this point would end up confusing. `Format` maybe less so.

Maybe `storage.Scheme`? We do have `RedundancyScheme`, but I think these are unlikely to end up near each other, because the storage node doesn't know anything about it.

Reply:
Ah, here it is. Yeah, "scheme" is a better word than "format", but I still would want to qualify it as a "storage scheme" in places where there could conceivably be doubt about what "scheme" means, and I would still want "version" as a modifier because this is meant to differentiate between multiple standardized, numbered points in storage-scheme-space. Multiple storage schemes, each having a version number.